### DISCUSSION TITLE
Bottles on main repo instead of Tap

### DISCUSSION THREAD

**Discussion by ronakfof** [Homebrew/discussions#3094] [Everyday usage]
Hi everyone,

I created a workflow for bottle creation on main repo. It would be great if you can check it out once. If there are no edge cases, It might be helpful to others.
PR - https://github.com/awslabs/aws-viewer-for-cbmc/pull/91

> This workflow can be used to `bump formula` and `generate bottles` for any `Formula` in any `Tap` and upload bottles to the corresponding release of main repo.
> 
> _Description of changes:_ On release, we create a pull request to bump brew formula to release version. With this PR, we checkout that pull request and run -
> 
>     1. Syntax Check
> 
>     2. Test installation
> 
>     3. Create bottle
> 
>     4. Test installation from bottle
> 
>     5. Test uninstallation
> 
>     6. Run test in formula
> 
>     7. Add commit on top of PR to add generate DSL(bottle stanzas) to formula
> 
> 
> **SAMPLE RUN**
> [Workflow run](https://github.com/ronakfof/viewer/actions/runs/2019789470)
> [Release](https://github.com/ronakfof/viewer/releases/tag/viewer-2.33)
> [Pull Request to Tap](https://github.com/ronakfof/homebrew-tap/pull/19/files)

**Comment by SMillerDev** [comment]
I don't think there are any brew commands that can do this since most of the work for this within Homebrew is also done by GitHub actions scripts

**Reply by ronakfof** [reply]
Hi,
I created a workflow for bottle creation on main repo. It would be great if you can check it out once. If there are no edge cases, It might be helpful to others.
PR - https://github.com/awslabs/aws-viewer-for-cbmc/pull/91

> This workflow can be used to `bump formula` and `generate bottles` for any `Formula` in any `Tap` and upload bottles to the corresponding release of main repo.
> 
> _Description of changes:_ On release, we create a pull request to bump brew formula to release version. With this PR, we checkout that pull request and run -
> 
>     1. Syntax Check
> 
>     2. Test installation
> 
>     3. Create bottle
> 
>     4. Test installation from bottle
> 
>     5. Test uninstallation
> 
>     6. Run test in formula
> 
>     7. Add commit on top of PR to add generate DSL(bottle stanzas) to formula
> 
> 
> **SAMPLE RUN**
> [Workflow run](https://github.com/ronakfof/viewer/actions/runs/2019534539)
> [Release](https://github.com/ronakfof/viewer/releases/tag/viewer-2.28)
> [Pull Request to Tap](https://github.com/ronakfof/homebrew-tap/pull/17/files)

**Reply by ronakfof** [reply]
@carlocab

**Comment by ronakfof** [comment]
Thank you for your response.
Is there a command to generate DSL for multiple bottles by passing multiple json files?

**Comment by ronakfof** [comment]
I am trying with - 
```
brew bottle --root-url $URL --merge --write --no-commit $(ls *.json)
```
But it's not adding the `root_url` to block

**Reply by gromgit** [reply]
IIRC, you need to use `--root-url` when building the bottle. The JSON must already contain the root URL during the merge.

**Reply by gromgit** [reply]
Also, is there a reason you need `$(ls *.json)` instead of just `*.json`?

**Reply by ronakfof** [reply]
> IIRC, you need to use --root-url when building the bottle. The JSON must already contain the root URL during the merge.

Thanks.

> Also, is there a reason you need $(ls *.json) instead of just *.json?

No specific reason

**Reply by ronakfof** [reply]
Do checkout the workflow i mentioned above for this discussion.
